mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostClassificationModelParitySpec.scala

```diff
@@ -24,16 +24,10 @@ class XGBoostClassificationModelParitySpec extends SparkParityBase {
   }
 
   val xgboostParams = Map(
-    "eta" -> 0.3,
-    "maxDepth" -> 2,
     "objective" -> "binary:logistic",
-    "treeMethod" -> "approx",
-    "earlyStoppingRounds" -> 2,
-    "numRound" -> 15,
-    "numClasses" -> 2,
-    "nWorkers" -> 2,
+    "num_classes" -> 2,
     "missing" -> 0.0f,
-    "allowNonZeroForMissing" -> true
+    "allow_non_zero_for_missing" -> true,
   )
```

Comment:
The xgboost estimator require param dict key using format like "xx_yy_zz", not "xxYyZz", so the master test all these params with capital letters are not set correctly.

I rewrite these param keys as format "xx_yy_zz". Note I don't add some params (e.g. max_depth) because those param don't be saved by model serializer so it will break the `serializes/deserializes the Spark model properly` test.  We can improve this in follow-up PR.

Reply:
I don't check detail code, but in my test, the camel style does not work, the parameter does not set correctly.
e.g. if we set `xgboostParams = Map("maxDepth" -> 2)` then the generated estimator the "maxDepth" is still 6 (default value) not 2.

Reply:
that's weird.  You're right that the xgboost scala tests are using snake case though.  So ok to leave it!